tests/test_processing_rt.py

```diff
@@ -0,0 +1,258 @@
+        axis_to_recalculate="z",
+    )
+    rt_xy_from_zx = rt_xz.isel(col=[0, 2, 1, 3])
+    rt_xy_from_zx[:, 2, :] = -rt_xy_from_zx[:, 2, :]
+
+    np.testing.assert_array_equal(rt_xy, rt_xy_x_recalc)
+    np.testing.assert_array_equal(rt_xy, rt_yx)
+    np.testing.assert_array_equal(rt_yz, rt_zy)
+    np.testing.assert_array_equal(rt_xz, rt_zx)
+    np.testing.assert_array_equal(rt_xy, rt_xy_from_yz)
+    np.testing.assert_array_equal(rt_xy, rt_xy_from_zx)
+
+    # Produce one that we know the solution
+    ref_m = Markers(np.array(((1, 2, 3), (4, 5, 6), (6, 5, 4))).T[:, :, np.newaxis])
+    rt_xy_from_known_m = Rototrans.from_markers(
+        origin=ref_m.isel(channel=[0]),
+        axis_1=ref_m.isel(channel=[0, 1]),
+        axis_2=ref_m.isel(channel=[0, 2]),
+        axes_name="xy",
+        axis_to_recalculate="y",
+    )
+
+    rt_xy_expected = Rototrans(
+        np.array(
+            [
+                [0.5773502691896257, 0.7071067811865475, -0.408248290463863, 1.0],
+                [0.5773502691896257, 0.0, 0.816496580927726, 2.0],
+                [0.5773502691896257, -0.7071067811865475, -0.408248290463863, 3.0],
+                [0, 0, 0, 1.0],
+            ]
+        )
+    )
+
+    np.testing.assert_array_equal(rt_xy_from_known_m, rt_xy_expected)
+
+    exception_default_params = dict(
+        origin=all_m.isel(channel=[0]),
+        axis_1=all_m.isel(channel=[0, 1]),
+        axis_2=all_m.isel(channel=[0, 2]),
+        axes_name="xy",
+        axis_to_recalculate="y",
+    )
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(origin=all_m.isel(channel=[0, 1]))}
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(axis_1=all_m.isel(channel=[0]))}
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(axis_2=all_m.isel(channel=[0]))}
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{
+                **exception_default_params,
+                **dict(axis_1=all_m.isel(channel=[0, 1], time=slice(None, 50))),
+            }
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(**{**exception_default_params, **dict(axes_name="yyz")})
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(**{**exception_default_params, **dict(axes_name="xxz")})
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(**{**exception_default_params, **dict(axes_name="zzz")})
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(axis_to_recalculate="h")}
+        )
+
+
+def test_rt_transpose():
+    n_frames = 10
+    angles = Angles.from_random_data(size=(3, 1, n_frames))
+    rt = Rototrans.from_euler_angles(angles, angle_sequence="xyz")
```

Comment:
from_random should be used as it should return a valid but random RotoTrans

Reply:
Done in https://github.com/pyomeca/pyomeca/pull/88/commits/319c128d37860b52cb4fc3e4cf9eefbaf36edc98